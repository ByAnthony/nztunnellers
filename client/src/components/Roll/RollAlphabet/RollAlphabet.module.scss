@use 'src/index.scss';

.roll {
  padding-bottom: index.$m-margin;

  .letter-container {
    width: 52px;
    background-color: index.$primary-dark-color;
    border: 1px solid index.$primary-light-color;
    border-radius: index.$border-radius;

    @include index.tablet {
      width: 62px;
    }

    @include index.desktop {
      width: 72px;
    }

    .title {
      margin: 0;
      padding: 8px 0 6px 0;
      font-size: 2rem;
      font-weight: 700;
      color: index.$text-color;
      text-align: center;

      @include index.tablet {
        font-size: 2.5rem;
      }

      @include index.desktop {
        font-size: 3rem;
      }
    }
  }

  .tunnellers-container {
    margin: index.$s-margin 0;
    display: flex;
    flex-direction: column;
    gap: index.$xs-margin;
  }

  @include index.desktop {
    width: 60%;
  }
}
