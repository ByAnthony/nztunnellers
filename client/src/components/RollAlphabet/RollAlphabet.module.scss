@use 'src/index.scss';

.roll {
  @include index.component-setup;
}

.letter-container {
  margin: 50px 0;
  width: 55px;
  background-color: index.$primary-medium-color;
  border: 1px solid index.$primary-light-color;
  border-radius: index.$border-radius;

  @include index.tablet {
    width: 65px;
  }

  @include index.desktop {
    margin: 75px 0;
    width: 90px;
  }
}

.letter-title {
  margin: 0;
  padding: 10px 0 7px 0;
  font-size: 2rem;
  font-weight: 700;
  color: index.$text-color;
  text-align: center;

  @include index.tablet {
    font-size: 2.5rem;
  }

  @include index.desktop {
    padding: 12px 0 5px 0;
    font-size: 4rem;
  }
}

.tunnellers-container {
  display: flex;
  flex-direction: column;
  gap: 15px;

  @include index.desktop {
    display: grid;
    gap: 25px;
    grid-template-columns: 1fr 1fr;
  }
}
