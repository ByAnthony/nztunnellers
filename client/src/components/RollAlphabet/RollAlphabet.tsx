import { Details } from '../../types/roll';
import { RollDetails } from '../RollDetails/RollDetails';
import STYLES from './RollAlphabet.module.scss';

type RollWithFilter = {
    tunnellers: Record<string, Details[]>;
    filterByLetter: string;
};

export function RollAlphabet({ tunnellers, filterByLetter }: RollWithFilter) {
  const tunnellersList = Object.entries(tunnellers);

  const isFilteredByLetter = (letter: string) => (letter === '' ? tunnellersList : tunnellersList.filter((key) => key.includes(letter)));

  const companyRoll = isFilteredByLetter(filterByLetter)
    .map(([key, listOfTunnellers]) => (
      <div id={`letter-${key}`} key={key}>
        <div className={STYLES['letter-container']}>
          <h2 className={STYLES.title} key={key} aria-label={`Letter ${key}`}>{key}</h2>
        </div>
        <div className={STYLES['tunnellers-container']}>
          <RollDetails listOfTunnellers={listOfTunnellers} />
        </div>
      </div>
    ));

  return (
    <div className={STYLES.roll}>
      { companyRoll }
    </div>
  );
}
