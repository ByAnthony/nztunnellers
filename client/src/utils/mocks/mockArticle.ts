import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
