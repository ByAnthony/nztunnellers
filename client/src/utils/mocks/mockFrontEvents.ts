import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
