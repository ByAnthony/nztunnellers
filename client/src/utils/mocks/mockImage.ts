import {
  Image, ImageSource, ImageNewspaper, ImageArchives, ImageBook,
} from '../../types/tunneller';

export const mockImageAucklandLibraries: string = 'https://digitalnz.org/records?text=31-B2671&tab=Images#';

export const mockImageArchives: ImageArchives = {
  location: 'Auckland War Memorial Museum',
  reference: 'MS-93/157',
};

export const mockImageFamily: string = 'Courtesy of John Doe family';

export const mockImageNewspaper: ImageNewspaper = {
  name: 'Auckland Star',
  date: '12 July 1898',
};

export const mockImageBook: ImageBook = {
  title: 'My Mock Title',
  town: 'Bethune',
  publisher: 'Publisher and Co.',
  authors: [
    {
      forename: 'Jane',
      surname: 'Doe',
    },
  ],
  year: '1925',
  page: 'p. 89',
};

export const mockImageSource: ImageSource = {
  aucklandLibraries: null,
  archives: null,
  family: null,
  newspaper: null,
  book: null,
};

export const mockImage: Image = {
  url: '1/1234.jpg',
  source: mockImageSource,
};
