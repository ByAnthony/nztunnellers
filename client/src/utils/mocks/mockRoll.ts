import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails = {
  id: mockId,
  name: johnDoe,
  birth: '1886',
  death: '1952',
};

export const mockRoll = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
