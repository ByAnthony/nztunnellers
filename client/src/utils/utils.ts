export const displayBirthDeathDates = (
  birth: string | null,
  death: string | null,
) => {
  if (birth !== null && death !== null) {
    return `${birth} - ${death}`;
  }
  if (birth !== null && death === null) {
    return `${birth} - †?`;
  }
  if (birth === null && death !== null) {
    return `? - ${death}`;
  }
  return '? - †?';
};
