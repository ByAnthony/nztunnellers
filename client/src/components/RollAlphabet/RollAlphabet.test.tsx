import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderWithMemoryRouter } from '../../utils/test-utils';
import { mockRoll } from '../../utils/mocks/mockRoll';
import { RollAlphabet } from './RollAlphabet';

const component = <RollAlphabet tunnellers={mockRoll} filterByLetter="" />;

test('renders the component correctly', () => {
  const { asFragment } = renderWithMemoryRouter(component);

  expect(asFragment()).toMatchSnapshot();
});

test('should filter by name', () => {
  renderWithMemoryRouter(<RollAlphabet tunnellers={mockRoll} filterByLetter="D" />);

  expect(screen.getByLabelText('Letter D')).toBeInTheDocument();
  expect(screen.queryByLabelText('Letter L')).not.toBeInTheDocument();
  expect(screen.queryByLabelText('Letter R')).not.toBeInTheDocument();

  expect(screen.getByText('John')).toBeInTheDocument();
  expect(screen.getByText('Doe')).toBeInTheDocument();

  expect(screen.getByText('Alexander')).toBeInTheDocument();
  expect(screen.getByText('Driscott')).toBeInTheDocument();

  expect(screen.queryByText('Robert')).not.toBeInTheDocument();
  expect(screen.queryByText('Lang')).not.toBeInTheDocument();

  expect(screen.queryByText('William')).not.toBeInTheDocument();
  expect(screen.queryByText('Right')).not.toBeInTheDocument();
});
