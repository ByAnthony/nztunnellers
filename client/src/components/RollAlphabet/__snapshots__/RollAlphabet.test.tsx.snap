// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="roll"
  >
    <div
      id="letter-D"
    >
      <div
        class="letter-container"
      >
        <h2
          aria-label="Letter D"
          class="title"
        >
          D
        </h2>
      </div>
      <div
        class="tunnellers-container"
      >
        <a
          class="tunneller-link"
          href="/tunnellers/26"
        >
          <div
            class="tunneller"
          >
            <div>
              <p
                class="forename"
              >
                John
              </p>
              <p
                class="surname"
              >
                Doe
              </p>
              <p
                class="dates"
              >
                1886 - 1952
              </p>
            </div>
            <div
              class="arrow-right"
            >
              →
            </div>
          </div>
        </a>
        <a
          class="tunneller-link"
          href="/tunnellers/1"
        >
          <div
            class="tunneller"
          >
            <div>
              <p
                class="forename"
              >
                Alexander
              </p>
              <p
                class="surname"
              >
                Driscott
              </p>
              <p
                class="dates"
              >
                1886 - 1952
              </p>
            </div>
            <div
              class="arrow-right"
            >
              →
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      id="letter-L"
    >
      <div
        class="letter-container"
      >
        <h2
          aria-label="Letter L"
          class="title"
        >
          L
        </h2>
      </div>
      <div
        class="tunnellers-container"
      >
        <a
          class="tunneller-link"
          href="/tunnellers/2"
        >
          <div
            class="tunneller"
          >
            <div>
              <p
                class="forename"
              >
                Robert
              </p>
              <p
                class="surname"
              >
                Lang
              </p>
              <p
                class="dates"
              >
                1886 - 1952
              </p>
            </div>
            <div
              class="arrow-right"
            >
              →
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      id="letter-R"
    >
      <div
        class="letter-container"
      >
        <h2
          aria-label="Letter R"
          class="title"
        >
          R
        </h2>
      </div>
      <div
        class="tunnellers-container"
      >
        <a
          class="tunneller-link"
          href="/tunnellers/3"
        >
          <div
            class="tunneller"
          >
            <div>
              <p
                class="forename"
              >
                William
              </p>
              <p
                class="surname"
              >
                Right
              </p>
              <p
                class="dates"
              >
                1886 - 1952
              </p>
            </div>
            <div
              class="arrow-right"
            >
              →
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
